core/global/launch_manager.gd

```diff
@@ -108,9 +108,21 @@ func _init() -> void:
 			# If the app has a gamepad profile, set it
 			if self._current_app:
 				set_app_gamepad_profile(self._current_app)
-
 		_xwayland_primary.focused_app_updated.connect(on_focused_app_changed)
 
+		# Listen for when focusable apps change
+		var on_focusable_apps_changed := func(from: PackedInt64Array, to: PackedInt64Array):
+			if from == to:
+				return
+			logger.debug("Focusable apps changed from", from, "to", to)
+			# If focusable apps has changed and the currently focused app no longer exists,
+			# remove the manual focus
+			var baselayer_app := _xwayland_primary.baselayer_app
```

Comment:
What is a base layer app and how is it defined?

Reply:
The baselayer app is a gamescope property (`GAMESCOPECTRL_BASELAYER_APP`) that can be set with an app id to manually make gamescope focus a specific app.